### ISSUE TITLE
Cloudwatch Event Rule not triggering?

### ISSUE THREAD

**Issue by Techn1x** [Signiant/dynamodb-autoscaling-to-slack#2]
I was really excited to implement this as practice for me so I can later attempt to implement some of the improved DynamoDB autoscaling implementations, I've hit a bit of a snag though, hoping someone can let me know if I'm doing the right thing

I created the CloudFormation using the template provided here in this project, that all created successfully. I left the TableFilter param blank, because I would like to be notified of all table updates (is this correct?). I can see the CloudWatch Event Rule, and the Lambda function, that were created.

I then place a load for 5+ minutes on my DynamoDB table, in the same region, and I can see the autoscaler goes ahead and scales as expected.

Looking at the metrics for the "DynamoDBAutoscaleUpdateTableEvent" event rule that was created, it's never being triggered (and subsequently neither is the lambda function). The event rule is set up as expected:
```json
{
  "source": [
    "aws.dynamodb"
  ],
  "detail": {
    "eventName": [
      "UpdateTable"
    ],
    "userAgent": [
      "application-autoscaling.amazonaws.com"
    ]
  }
}
```

Is there anything more I need to set up? Anything I should check?

-----

Not sure if this is related, but I went to create my own Event Rule for troubleshooting, and it allows me to select DynamoDB for the service/source, but then there are no Event Types available (only one called "AWS API Call via CloudTrail"). This doc mentions this (see point 3c), but not what it means or how to fix it https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/Create-CloudWatch-Events-Rule.html
![untitled24](https://user-images.githubusercontent.com/1049837/46999444-0cb8a980-d169-11e8-94a0-ceb99c30906b.png)

Thanks in advance, really scratching my head over this one.

**Comment by dnorth98** [comment]
Very strange.  Yeah leaving the table filter blank should be fine.  As long as you're not seeing this in the log for the lambda

`"Table` filtered: " + table_name

then it shouldn't be ignoring the table.

One thing I DID just think of though based on your last note is do you have Cloudtrail enabled?  If not, I don't think the UpdateTable call will be tracked as an event as I believe cloudwatch events use Cloudtrail to trigger on an API call.  Thinking more about it, I bet this is it....

**Comment by Techn1x** [comment]
Thanks for the reply, I'll do some experimenting with CloudTrail and see what I can find! I didn't dive too deep into CloudTrail at first because it felt like a red herring (I was under the impression based on the Event Rule definition that the events went straight from DDB to CloudWatch - but this assumption may have been incorrect)

I have no Trails set up, but the UpdateTable events do show in the "Event History" part of CloudTrail
![untitled25](https://user-images.githubusercontent.com/1049837/47193878-79fe5180-d398-11e8-9e9f-0cc86b77d97d.png)

Now that I know to look at CloudTrail I can do a bit more sleuthing

**Comment by Techn1x** [comment]
Managed to get it working. I had to do two things;

1) Enable CloudTrail - it seems that just having the on-by-default 90-day Event History part of CloudTrail isn't enough. I also needed a Trail created that logged "Management events" (specifically logging at least 'write' events, which includes UpdateTable operation)

2) In the CloudWatch Event Rule, I had to change the `userAgent` from `application-autoscaling.amazonaws.com` to `dynamodb.application-autoscaling.amazonaws.com` - not sure why this might be different for me, and if this works for others or not

Would you like changes 1) and/or 2) in the CloudFormation script? I'm happy to do a PR if you like.

**Comment by dnorth98** [comment]
Weird.  Yes please do send a PR.  I updated the readme yesterday about requiring CT being enabled but perhaps you could add that small nugget too.

Really glad you were able to get it to work!

**Comment by Techn1x** [comment]
PR here https://github.com/Signiant/dynamodb-autoscaling-to-slack/pull/3

Didn't end up adding CloudTrail to the CloudFormation setup since it's fairly straightforward to setup and I think the readme entry you added covers it. Most users probably already have existing trails setup anyway!

Thanks for the help @dnorth98

**Comment by dnorth98** [comment]
Merged in PR so this is all fixed in  1e64720.  Thanks again!!
